bitnami/postgresql-ha/templates/postgresql/statefulset.yaml

```diff
@@ -189,7 +189,7 @@ spec:
               value: {{ .Values.persistence.mountPath | quote }}
             - name: PGDATA
               value: {{ printf "%s/%s" .Values.persistence.mountPath "data" | quote }}
-            {{- if and (not (eq (include "postgresql-ha.postgresqlUsername" .) "postgres")) (or (not (include "postgresql-ha.postgresqlCreateSecret" .)) (include "postgresql-ha.postgresqlPasswordProvided" .)) }}
+            {{- if and (not (eq (include "postgresql-ha.postgresqlUsername" .) "postgres")) (or ((include "postgresql-ha.postgresqlCreateSecret" .)) (include "postgresql-ha.postgresqlPasswordProvided" .)) }}
```

Comment:
I need to further investigate this issue, but there is currently bug either on this line, where the logic should be simplified as:
```
{{- if and (not (eq (include "postgresql-ha.postgresqlUsername" .) "postgres")) }}
```

In this line:
https://github.com/bitnami/charts/blob/68b3f324b6b96947cf232b9f127677dbbb264c07/bitnami/postgresql-ha/templates/postgresql/secrets.yaml#L19
Where the condition to create the postgres-password is always true

Or in the container logic, where container initialization will fail with the following error if password for `postgres` user is not provided when a different user is used:
```
postgresql-repmgr 16:01:42.52 INFO  ==> Creating repmgr user: repmgr
2024-10-31 16:01:42.544 GMT [233] FATAL:  password authentication failed for user "postgres"
2024-10-31 16:01:42.544 GMT [233] DETAIL:  User "postgres" has no password assigned.
        Connection matched file "/opt/bitnami/postgresql/conf/pg_hba.conf" line 8: "host     all              all       0.0.0.0/0    md5"
psql: error: connection to server at "127.0.0.1", port 5432 failed: FATAL:  password authentication failed for user "postgres"
```

Reply:
Yeah maybe this is not the correct fix but it seems something is strange indeed.